pkg/logs/sender/message_buffer.go

```diff
@@ -9,27 +9,34 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/logs/message"
 )
 
-// MessageBuffer accumulates messages to a buffer until the max capacity is reached.
+// MessageBuffer accumulates message metadata to a buffer until the max capacity is reached.
 type MessageBuffer struct {
-	messageBuffer    []*message.Message
+	messageBuffer    []*message.MessageMetadata
 	contentSize      int
 	contentSizeLimit int
 }
 
 // NewMessageBuffer returns a new MessageBuffer.
 func NewMessageBuffer(batchSizeLimit int, contentSizeLimit int) *MessageBuffer {
 	return &MessageBuffer{
-		messageBuffer:    make([]*message.Message, 0, batchSizeLimit),
+		messageBuffer:    make([]*message.MessageMetadata, 0, batchSizeLimit),
 		contentSizeLimit: contentSizeLimit,
 	}
 }
 
+// CanAddMessage returns true if the message can be added to the buffer.
+func (p *MessageBuffer) CanAddMessage(message *message.Message) bool {
```

Comment:
What is the purpose of moving the previous logic of `AddMessage` to `CanAddMessage`? Is it so `addMessage` can determine if it should serialize the message?

I'm just curious because it seems like `addMessage` will inadvertently call `CanAddMesage` twice.

Reply:
It so we can ensure a message is only considered added if it was successfully serialized. 
It's possible that:
- there is room in the buffer
- serialize fails

In this case, we would need to NOT add it to the buffer. This wasn't supported in the old API. Adding `CanAddMessage` allows us to check if it would fit before adding it to ensure this failure mode is handled.